apps/crn-server/src/data-providers/users.ts

```diff
@@ -0,0 +1,338 @@
+import { GenericError } from '@asap-hub/errors';
+import { UserDataObject, UserPatchDataObject } from '@asap-hub/model';
+import {
+  config,
+  RestUser,
+  sanitiseForSquidex,
+  SquidexGraphqlClient,
+  SquidexRest,
+  SquidexRestClient,
+} from '@asap-hub/squidex';
+import Intercept from 'apr-intercept';
+import FormData from 'form-data';
+import { Got } from 'got';
+import mime from 'mime-types';
+import {
+  FetchUserQuery,
+  FetchUserQueryVariables,
+  FetchUsersQuery,
+  FetchUsersQueryVariables,
+} from '../autogenerated-gql/graphql';
+import { FetchUsersOptions } from '../controllers/users';
+import {
+  parseGraphQLUserToDataObject,
+  parseUserToDataObject,
+} from '../entities';
+import { FETCH_USER, FETCH_USERS } from '../queries/users.queries';
+import { fetchOrcidProfile, transformOrcidWorks } from '../utils/fetch-orcid';
+
+export interface UserDataProvider {
+  fetchById(id: string): Promise<UserDataObject | null>;
+  update(id: string, update: UserPatchDataObject): Promise<void>;
+  fetch(options: FetchUsersOptions): Promise<UserDataObject[]>;
+  fetchByCode(code: string): Promise<UserDataObject[]>;
+  updateAvatar(id: string, avatar: Buffer, contentType: string): Promise<void>;
+  connectByCode(
+    welcomeCode: string,
+    userId: string,
+  ): Promise<UserDataObject | null>;
+  syncOrcidProfile(
+    id: string,
+    cachedUser?: RestUser | undefined,
+  ): Promise<UserDataObject>;
+}
+export default function createUserDataProvider(
+  squidexGraphlClient: SquidexGraphqlClient,
+): UserDataProvider {
+  const userSquidexRestClient = new SquidexRest<RestUser>('users');
+  const fetchById = async (id: string) => {
+    const { findUsersContent } = await queryFetchByIdData(
+      squidexGraphlClient,
+      id,
+    );
+    if (!findUsersContent) {
+      return null;
+    }
+    return parseGraphQLUserToDataObject(findUsersContent);
+  };
+  const update = async (id: string, userToUpdate: UserPatchDataObject) => {
+    const isFullUpdate = shouldDoFullUpdate(userToUpdate);
+
+    const cleanedUser = cleanUser(userToUpdate);
+
+    if (isFullUpdate) {
+      const existingUser = await userSquidexRestClient.fetchById(id);
+      await userSquidexRestClient.put(id, {
+        ...existingUser.data,
+        ...cleanedUser,
+      });
+    } else {
+      await userSquidexRestClient.patch(id, cleanedUser);
+    }
+  };
+  const fetch = async (options: FetchUsersOptions) => {
+    const queryFilter = generateFetchQueryFilter(options);
+    const { take = 8, skip = 0 } = options;
+    const users = await queryForUsers(
+      squidexGraphlClient,
+      queryFilter,
+      take,
+      skip,
+    );
+    return users.map(parseGraphQLUserToDataObject);
```

Comment:
you should return here also the total from the query response

Reply:
The total is added in by the controller.